src/CommonDBTM.php

```diff
@@ -3141,6 +3141,29 @@ public function check($ID, int $right, ?array &$input = null): void
         }
     }
 
+    /** @param int[] $entities_ids */
+    public function isAccessibleFromEntities(array $entities_ids): bool
+    {
+        if (!$this->isEntityAssign()) {
+            // Item does not have any entity so it is always visible.
+            return true;
+        }
+
+        if ($this->maybeRecursive() && $this->fields['is_recursive']) {
+            // Item is recursive, check if it is accessible from any of the ancestors of the given entities.
+            return in_array(
+                $this->getEntityID(),
+                getAncestorsOf("glpi_entities", $entities_ids),
+            );
+        } else {
+            // Item is not recursive, check if it is accessible from any of the given entities.
+            return in_array(
+                $this->getEntityID(),
+                $entities_ids,
+            );
+        }
```

Comment:
You can optimize this operation by first checking if `in_array($this->getEntityID(), $entities_ids,)`. In many cases, it may return `true` and would prevent having to fetch the ancestors in the cache (filesystem I/O) or the DB.